src/iop/basecurve.c

```diff
@@ -516,6 +522,13 @@ int process_cl_fusion(struct dt_iop_module_t *self, dt_dev_pixelpipe_iop_t *piec
   cl_mem dev_m = NULL;
   cl_mem dev_coeffs = NULL;
 
+  const int use_work_profile = (work_profile == NULL) ? 0 : 1;
+  const int preserve_colors = d->preserve_colors;
+
+  err = dt_ioppr_build_iccprofile_params_cl(work_profile, devid, &profile_info_cl, &profile_lut_cl,
```

Comment:
You need to call dt_ioppr_free_iccprofile_params_cl() to match this.

Reply:
Thanks!  Good catch, I can easily fix that tonight.

One question:  Since github has the ability to make a rebase/squash/force-repush of a pull request less painful for all reviewers, is that acceptable for this fix?  Doing a rebase/squash has the benefit of keeping the commit history cleaner/more concise for someone trying to follow history down the line.

I've noticed people have tended towards lots of "fix" commits without a squash prior to merge, that sometimes makes the history of the code quite difficult to read - but is that an artifact of when darktable lived elsewhere other than github?

Reply:
Not really my area, I always try to have one commit per functionality, but is not always the case.

Reply:
That's generally my goal, but I'm not sure if there's a policy against rebasing/squashing an amended pull request.  Some people really dislike it (even if it does in the long run result in cleaner commits)

Linus hates it, but the email-based kernel development flow pretty much amounts to an alternative to rebases/squashes within git, by resending a patch set as emails instead.

Reply:
I pushed an amended commit that frees those data structures